### PR TITLE
clientv3: Fix new load balancer integration issues

### PR DESCRIPTION
**Pull request by jpbetz**
A couple changes here:
- Remove Endpoint tracking from the new balancer implementation entirely, since we don't seem to need it
- Register the balancer statically in an `init()` function since it no longer has any per-client state and registering it on the fly triggered https://github.com/grpc/grpc-go/issues/1973
- Fix a bug where a target was passed in where a endpoint was expected

With this in place, the only failing tests are in `github.com/coreos/etcd/clientv3` and `github.com/coreos/etcd/clientv3/ordering`.

Example `github.com/coreos/etcd/clientv3` failure:

```
--- FAIL: TestDialTimeout (6.34s)
         client_test.go:102: #0: new client should fail
         client_test.go:111: #0: dial didn't wait (<nil>)
         client_test.go:117: #0: failed to timeout dial on time
         leak.go:107: Test appears to have leaked a timing out dial:
           github.com/coreos/etcd/vendor/google.golang.org/grpc.(*ccBalancerWrapper).watcher(0xc420010a80)
             /usr/local/google/home/jpbetz/Projects/etcd/src/github.com/coreos/etcd/vendor/google.golang.org/grpc/balancer_conn_wrappers.go:122 +0x1e7
           created by github.com/coreos/etcd/vendor/google.golang.org/grpc.newCCBalancerWrapper
             /usr/local/google/home/jpbetz/Projects/etcd/src/github.com/coreos/etcd/vendor/google.golang.org/grpc/balancer_conn_wrappers.go:113 +0x2f2
```

Example `github.com/coreos/etcd/clientv3/ordering` failure:

```
--- FAIL: TestUnresolvableOrderViolation (5.53s)
         util_test.go:150: expected etcdclient: no cluster members have a revision higher than the previously received revision, got rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc= "transport: Error while dialing dial unix localhost:33144138729382583520: connect: no such file or directory"
```

### CODE REVIEW

File: clientv3/client.go
Comment (gyuho) [inline clientv3/client.go]: `Name: picker.RoundrobinBalanced.String(),` or define `rrbalancer` as `const`?

**Comment by gyuho** [comment]
@jpbetz Just one nit, and lgtm.

Also, the new gRPC with unix socket should be released by today.
In next commits, we can clean up the manual patch on gRPC and also
address other test failures. I will also try to reproduce them, once merged.

Thanks!

File: clientv3/client.go
Comment (jpbetz) [reply clientv3/client.go]: Since we're registering a grpc globally in a process where the etcd client might not be the only thing grpc is used for, I'll scope it-- `fmt.Sprintf("etcd-%s", picker.RoundrobinBalanced.String())`.

**Comment by jpbetz** [comment]
@gyuho I've updated to grpc 1.11.3. The version bump is in a separate "vendor: upgrade "grpc/grpc-go" to v1.11.3" commit.

**Comment by jpbetz** [comment]
Ohh, don't merge yet, got a failure in `./test` after the grpc version bump:

```
'markdown_you' started at Mon Apr  9 14:17:23 PDT 2018                                                                                                                                                                              
found 'you' in documentation:                                                                                                                               
./vendor/google.golang.org/grpc/CONTRIBUTING.md:We definitely welcome your patches and contributions to gRPC!                                                      

```

File: test
Comment (jpbetz) [inline test:365]: @gyuho We're okay with excluding `/vendor`, right?  I've also switched to `! -path` style filtering here since it's already used elsewhere in this test file and it make the file more consistent.

**Review by gyuho — APPROVED** [review]
awesome. lgtm. thanks!

File: test
Comment (gyuho) [reply test:365]: @jpbetz Yes. I tried your command, and confirm that it fixes. We want to exclude `vendor`. Seems like it's been broken since :0

**Pull request merged**
